### DISCUSSION TITLE
File Storage with OpenStack Swift

### DISCUSSION THREAD

**Discussion by Silbad** [directus/directus#14376] [Archive]
Hello,
Is it possible to use openstack swift to store files as it was possible with Directus 8?
https://v8.docs.directus.io/docker/environment.html#all-variables
Thanks

**Comment by rijkvanzanten** [comment]
Looks like V8 recommended to use the S3 adapter to connect to OpenStack Swift, so if that tool is s3 compatible, you should be able to use the regular S3 driver 👍🏻

**Reply by Silbad** [reply]
oh ok I can try :) thanks.

**Reply by Silbad** [reply]
It works like a charm !

**Reply by artadam** [reply]
@Silbad Hi! I am trying to do it but got S3ServiceException "The specified bucket is not valid.". Here is my config:

STORAGE_LOCATIONS="OPENSTACK,local"
STORAGE_OPENSTACK_DRIVER="s3"
STORAGE_OPENSTACK_KEY="<OS_APPLICATION_CREDENTIAL_ID>"
STORAGE_OPENSTACK_SECRET="<OS_APPLICATION_CREDENTIAL_SECRET>"
STORAGE_OPENSTACK_REGION="us-east-1"
STORAGE_OPENSTACK_ENDPOINT="http://172.16.105.10:8080/v1/AUTH_199a539baaef40daaf7d2fe06238a9d0"
STORAGE_OPENSTACK_BUCKET="lobiji-test"
STORAGE_OPENSTACK_FORCE_PATH_STYLE="true"

as bucket i defined openstack container name and 199a539baaef40daaf7d2fe06238a9d0 is project id.
My swift objectstore region is "Riga", but it do not allow me to define it

May be i do smth wrong?
Can you share configuration example please?

**Comment by Silbad** [comment]
Hello, mm it's an old post.
Go to https://directus.io/docs/configuration/files
STORAGE_OPENSTACK does not exist.
You have only  local, s3, gcs, azure, cloudinary, supabase.
So IF S3 is compatible, change by STORAGE_S3_SECRET, ...
++
